src/main/fc/fc_core.c

```diff
@@ -485,7 +483,9 @@ void processRx(timeUs_t currentTimeUs)
             DISABLE_FLIGHT_MODE(HEADFREE_MODE);
         }
         if (IS_RC_MODE_ACTIVE(BOXHEADADJ)) {
-            headFreeModeHold = DECIDEGREES_TO_DEGREES(attitude.values.yaw); // acquire new heading
```

Comment:
Only difference between [turning off HEADFREE mode, flying, reenabling HEADFREE] and [enabling BOXHEADADJ, flying, disabling BOXHEADADJ] which user can detect is reference orientation after sequence - it is remembered from ARM in HEADFREE case and reset in BOXHEADADJ. Everything other is the same.